### PR TITLE
OSOE-678: Upgrade to Stylelint 15 and migrate style linting rules to Prettier

### LINKED ISSUE #70: Upgrade to Stylelint 15 and handle breaking changes (OSOE-678)
Followup to https://github.com/Lombiq/NodeJs-Extensions/issues/67#issuecomment-1692126657

> Updating stylelint to 15.10.3 from 14.14.0 would mean a major update that contains the following (breaking) changes:
[https://stylelint.io/migration-guide/to-15](https://stylelint.io/migration-guide/to-15)
>
>To sum up, they deprecated 76 rules because they are mostly styling related rules that can be achieved by pretty printers like Prettier so they won't support it anymore from the next major update.
>
>These are the rules that we are using and deprecated:
>```
>'declaration-block-semicolon-newline-before': 'never-multi-line',
>'function-comma-newline-after': 'always-multi-line',
>'indentation': 4,
>'max-line-length': 150,
>'selector-list-comma-space-after': 'always-single-line',
>'selector-list-comma-newline-after': 'always-multi-line',
>'value-list-comma-newline-before': 'never-multi-line'
>```

Consider how to handle these breaking rules and evaluate if using Prettier is a good fit.

[Jira issue](https://lombiq.atlassian.net/browse/OSOE-678)

### PR DESCRIPTION
**Pull request by wAsnk**
[OSOE-678](https://lombiq.atlassian.net/browse/OSOE-678)
Fixes #70

[OSOE-678]: https://lombiq.atlassian.net/browse/OSOE-678?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: Lombiq.NodeJs.Extensions/config/.stylelintrc.lombiq-base.js [hunk 1]
```diff
@@ -1,20 +1,12 @@
...
-        'function-comma-newline-after': 'always-multi-line',
-        'indentation': 4,
         'keyframes-name-pattern': null,
-        'max-line-length': 150,
         'selector-class-pattern': null,
         'selector-id-pattern': null,
-        'selector-list-comma-space-after': 'always-single-line',
         'selector-no-qualifying-type': true,
-        'selector-list-comma-newline-after': 'always-multi-line',
-        'value-list-comma-newline-before': 'never-multi-line'
```
Comment (DemeSzabolcs) [inline Lombiq.NodeJs.Extensions/config/.stylelintrc.lombiq-base.js:18]: So PrettierStyle takes care of this?

File: Lombiq.NodeJs.Extensions/Docs/Styles.md [hunk 2]
```diff
@@ -60,26 +61,33 @@ Now, you can add any or all of the following entries to the `scripts` property i
...
 
 ```xml
 <NodeJsExtensionsGlobalStylelintConfigurationDirectory>$(MSBuildThisFileDirectory)</NodeJsExtensionsGlobalStylelintConfigurationDirectory>
+<NodeJsExtensionsGlobalPrettierStyleConfigurationDirectory>$(MSBuildThisFileDirectory)</NodeJsExtensionsGlobalPrettierStyleConfigurationDirectory>
 ```
 
-Details on rules can be found in the [Stylelint documentation](https://stylelint.io/user-guide/rules/list/). If you want to find out what the currently applied configuration is, coming from all the various extended configuration files, then run `npx stylelint --print-config . > rules.json` at the given location.
+Details on rules can be found in the [Stylelint documentation](https://stylelint.io/user-guide/rules/list/) and in the [Prettier documentation](https://prettier.io/docs/en/options). If you want to find out what the currently applied configuration is for Stylelint, coming from all the various extended configuration files, then run `npx stylelint --print-config . > rules.json` at the given location.
```
Comment (DemeSzabolcs) [inline Lombiq.NodeJs.Extensions/Docs/Styles.md:90]: > out what the currently applied configuration is for Stylelint, coming from all the various extended configuration files, then run `npx stylelint --print-config . > rules.json` at the given location.

Is there a similar thing for `Prettier`?

File: Lombiq.NodeJs.Extensions/build/CreateGlobalConfigFiles.targets [hunk 3]
```diff
@@ -31,6 +41,16 @@
           Condition="'@(StylelintrcFile)' == ''" />
   </Target>
 
+  <Target Name="_ProcessPrettierStyleConfigFile" DependsOnTargets="EnablePnpm">
+    <ItemGroup>
+      <PrettierrcFile Include="$(NodeJsExtensionsGlobalPrettierStyleConfigurationDirectory)/.prettierrc*" />
+    </ItemGroup>
+    <!-- Copy the global .stylelintrc.js to the desired directory. -->
```
Comment (DemeSzabolcs) [inline Lombiq.NodeJs.Extensions/build/CreateGlobalConfigFiles.targets]:
```suggestion
    <!-- Copy the global .prettierrc.js to the desired directory. -->
```

**Review by DemeSzabolcs — CHANGES_REQUESTED** [review]

File: Lombiq.NodeJs.Extensions/config/.stylelintrc.lombiq-base.js (hunk 1 above)
Comment (wAsnk) [reply Lombiq.NodeJs.Extensions/config/.stylelintrc.lombiq-base.js:18]: Yup. But you can check it how it is handled here https://prettier.io/playground/

File: Lombiq.NodeJs.Extensions/Docs/Styles.md (hunk 2 above)
Comment (wAsnk) [reply Lombiq.NodeJs.Extensions/Docs/Styles.md:90]: Currently there isn't. https://github.com/prettier/prettier/issues/15156

**Review by DemeSzabolcs — APPROVED** [review]

**Pull request merged**